caravel/views.py

```diff
@@ -1581,6 +1580,59 @@ def queries(self, last_updated_ms):
             mimetype="application/json")
 
     @has_access
+    @expose("/get_all_users")
+    @log_this
+    def all_users(self):
+        """Get all the queries."""
+        if not g.user.get_id():
+            return Response(
+                json.dumps({'error': "Please login to access the queries."}),
+                status=403,
+                mimetype="application/json")
+
+        return Response(
+            json.dumps([(u.id, u.username) for u in sm.get_all_users()]),
+            status=200,
+            mimetype="application/json")
+
+    @has_access
+    @expose("/search_queries/userId=<userId>"
+        "&dbId=<dbId>&sql=<search_text>&state=<state>")
+    @log_this
+    def search_queries(self, userId, dbId, search_text, state):
+        """Get all the queries."""
+        if not g.user.get_id():
+            return Response(
+                json.dumps({'error': "Please login to access the queries."}),
+                status=403,
+                mimetype="application/json")
+
+        search_str = '%{}%'.format(search_text)
+
+        query = db.session.query(models.Query)
+        if not userId == 'null':
```

Comment:
I believe sqlalchemy is fine with the None values, it just ignores them, your query could look like:
query = query.filter(
    models.Query.user_id == userId,
   models.Query.database_id == dbId, ...
@mistercrunch  - could you confirm that?